client/actions/events/ui.js

```diff
@@ -0,0 +1,119 @@
+ */
+const _openSpikeModal = (event) => (
+    (dispatch) => {
+        if (!event.recurrence_id) {
+            return dispatch(self._openSingleSpikeModal(event))
+        } else {
+            return dispatch(self._openMultiSpikeModal(event))
+        }
+    }
+)
+
+/**
+ * Open the Spike Single Modal
+ * @param {object} event - The Event to be spiked
+ */
+const _openSingleSpikeModal = (event) => (
+    (dispatch, getState, { notify }) => (
+        dispatch(planning.api.loadPlanningByEventId(event._id))
+        .then((planningItems) => (
+            dispatch(showModal({
+                modalType: 'SPIKE_EVENT',
+                modalProps: {
+                    eventDetail: {
+                        ...event,
+                        _plannings: planningItems,
+                    },
+                },
+            }))
+        ), (error) => {
+            notify.error(
+                getErrorMessage(error, 'Failed to load associated Planning items.')
+            )
+            return Promise.reject(error)
+        })
+    )
+)
+
+/**
+ * Open the Spike Multiple Modal
+ * @param {object} event - The Event to be spiked
+ */
+const _openMultiSpikeModal = (event) => (
+    (dispatch, getState, { notify }) => (
+        dispatch(eventsApi.loadRecurringEventsAndPlanningItems(event))
+        .then((events) => {
+            dispatch(showModal({
+                modalType: 'SPIKE_EVENT',
+                modalProps: { eventDetail: events },
+            }))
+            return Promise.resolve(events)
+        }, (error) => {
+            notify.error(
+                getErrorMessage(
+                    error,
+                    'Failed to load associated planning items'
+                )
+            )
+            return Promise.reject(error)
+        })
+    )
+)
+
+/**
+ * Spike an event and notify the user of the result
+ * @param {object} event - The event to spike
+ */
+const spike = (event) => (
+    (dispatch, getState, { notify }) => (
+        dispatch(eventsApi.spike(event))
+        .then((events) => {
+            dispatch(hideModal())
+            const currentPlanId = selectors.getCurrentPlanningId(getState())
+
+            events.forEach((e) => {
+                if (e._plannings) {
+                    const planIds = e._plannings.map((p) => (p._id))
+                    if (planIds.indexOf(currentPlanId) > -1) {
+                        dispatch(planning.ui.closeEditor())
+                    }
+                }
+            })
+
+            notify.success('The event(s) have been spiked')
+            return Promise.resolve(events)
+        }, (error) => {
+            notify.error(
+                getErrorMessage(error, 'Failed to spike the event(s)')
+            )
+
+            return Promise.reject(error)
+        })
+    )
+)
+
+const openSpikeModal = checkPermission(
+    _openSpikeModal,
+    PRIVILEGES.SPIKE_EVENT,
+    'Unauthorised to spike an Event'
+)
+
+const self = {
+    _openSpikeModal,
```

Comment:
I think we use `_*` names for internal used variables. If we export them, they are not internal anymore.
I guess they are exported for the tests and I found recently a nice module helping for accessing and modifying internal variables: https://github.com/jhnns/rewire
I don't know if it worth it... what do you think ?